src/Type/Php/ExplodeFunctionDynamicReturnTypeExtension.php

```diff
@@ -41,32 +40,34 @@ public function getTypeFromFunctionCall(
 		Scope $scope,
 	): ?Type
 	{
-		if (count($functionCall->getArgs()) < 2) {
+		$args = $functionCall->getArgs();
+		if (count($args) < 2) {
 			return null;
 		}
 
-		$delimiterType = $scope->getType($functionCall->getArgs()[0]->value);
-		$isSuperset = (new ConstantStringType(''))->isSuperTypeOf($delimiterType);
-		if ($isSuperset->yes()) {
+		$delimiterType = $scope->getType($args[0]->value);
+		$isEmptyString = (new ConstantStringType(''))->isSuperTypeOf($delimiterType);
+		if ($isEmptyString->yes()) {
 			if ($this->phpVersion->getVersionId() >= 80000) {
 				return new NeverType();
 			}
 			return new ConstantBooleanType(false);
-		} elseif ($isSuperset->no()) {
-			$arrayType = AccessoryArrayListType::intersectWith(new ArrayType(new IntegerType(), new StringType()));
-			if (
-				!isset($functionCall->getArgs()[2])
-				|| IntegerRangeType::fromInterval(0, null)->isSuperTypeOf($scope->getType($functionCall->getArgs()[2]->value))->yes()
-			) {
-				return TypeCombinator::intersect($arrayType, new NonEmptyArrayType());
-			}
+		}
+
+		$returnType = AccessoryArrayListType::intersectWith(new ArrayType(new IntegerType(), new StringType()));
+		if (
+			!isset($args[2])
+			|| IntegerRangeType::fromInterval(0, null)->isSuperTypeOf($scope->getType($args[2]->value))->yes()
+		) {
+			$returnType = TypeCombinator::intersect($returnType, new NonEmptyArrayType());
+		}
 
-			return $arrayType;
+		if ($this->phpVersion->getVersionId() <= 80000 && $isEmptyString->maybe()) {
```

Comment:
Do not use phpVersion like that, create a new method that describes the thing. Or use an existing method.